### ISSUE TITLE
sstables_directory: refuse loading sstables if orphaned sstables are found

### ISSUE THREAD

**Issue by denesb** [scylladb/scylladb#18038]
orphaned=belongs to a table which has no local replica
Such sstables can indicate a past race with a migration and ignoring them can cause data resurrection. So we should either auto-delete them, or refuse to start-up, letting this be done by the admin.

**Comment by denesb** [comment]
Maybe we already do this, we should write a dedicated test as a start, to see what happens.

**Comment by raphaelsc** [comment]
Yes, I think we would refuse to load such a sstable because no storage group would be found. Actually, this will be true after @Deexie's series on dynamic allocation

**Comment by raphaelsc** [comment]
But note there's theoretically a sequence of events, where tablet replica can go away and back, and we didn't have a chance to delete the sstable because there was no restart, whatever. When tablet is migrated back, and we restart, the old sstable can be loaded.

**Comment by raphaelsc** [comment]
We're relying on migration algorithm to always cleanup sstables, on success or failure.

**Comment by bhalevy** [comment]
As I mentioned in https://github.com/scylladb/scylladb/pull/17664#issuecomment-2020541859,
If an attempt to add the sstable to a tablet finds out it was already migrated away, we should unlink it on the spot.
If scylla crashes before that, then restart should find the sstable as orphaned and refuse to start.

**Comment by bhalevy** [comment]
This is a secondary issue and a theoretical concern, hence let's postpone it to 6.1

**Comment by raphaelsc** [comment]
> The case is already handled by the server and an error is thrown : [table.cc:659](https://github.com/scylladb/scylladb/blob/9024e4940cb827921f2d1b60e349676eab8788df/replica/table.cc#L659). I have added a testcase and hopefully that is enough to close this issue.

Do we decorate that message with the sstable name? So admin can act after sstable loading fails on boot?

**Comment by lkshminarayanan** [comment]
> > The case is already handled by the server and an error is thrown : [table.cc:659](https://github.com/scylladb/scylladb/blob/9024e4940cb827921f2d1b60e349676eab8788df/replica/table.cc#L659). I have added a testcase and hopefully that is enough to close this issue.
> 
> Do we decorate that message with the sstable name? So admin can act after sstable loading fails on boot?

@raphaelsc yes this is the error being thrown :

`Storage wasn't found for tablet {tablet-id} of table {keyspace}.{columnfamily}`

**Comment by raphaelsc** [comment]
> > > The case is already handled by the server and an error is thrown : [table.cc:659](https://github.com/scylladb/scylladb/blob/9024e4940cb827921f2d1b60e349676eab8788df/replica/table.cc#L659). I have added a testcase and hopefully that is enough to close this issue.
> > 
> > 
> > Do we decorate that message with the sstable name? So admin can act after sstable loading fails on boot?
> 
> [@raphaelsc](https://github.com/raphaelsc) yes this is the error being thrown :
> 
> `Storage wasn't found for tablet {tablet-id} of table {keyspace}.{columnfamily}`

I am actually looking for something like this:
```
        on_internal_error(tlogger, format("Unable to load SSTable {} that belongs to tablets {} and {}",
                                          sst->get_filename(), first_id, last_id));
```

perhaps we can catch exception thrown by storage_group_for_id() call in compaction_group_for_sstable() and decorate it with the sstable name (not table) as above.

**Comment by lkshminarayanan** [comment]
> > > > The case is already handled by the server and an error is thrown : [table.cc:659](https://github.com/scylladb/scylladb/blob/9024e4940cb827921f2d1b60e349676eab8788df/replica/table.cc#L659). I have added a testcase and hopefully that is enough to close this issue.
> > > 
> > > 
> > > Do we decorate that message with the sstable name? So admin can act after sstable loading fails on boot?
> > 
> > 
> > [@raphaelsc](https://github.com/raphaelsc) yes this is the error being thrown :
> > `Storage wasn't found for tablet {tablet-id} of table {keyspace}.{columnfamily}`
> 
> I am actually looking for something like this:
> 
> ```
>         on_internal_error(tlogger, format("Unable to load SSTable {} that belongs to tablets {} and {}",
>                                           sst->get_filename(), first_id, last_id));
> ```
> 
> perhaps we can catch exception thrown by storage_group_for_id() call in compaction_group_for_sstable() and decorate it with the sstable name (not table) as above.

@raphaelsc done.
